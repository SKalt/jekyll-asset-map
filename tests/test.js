/* eslint-env node */ /* eslint-env jest */
const cheerio = require("cheerio");
const fs = require("fs");
const path = require("path");

const testIds = [
  "path_present",
  "path_missing",

  "integrity",
  "integrity_missing",

  "error_alert",
  "error_error",
  "error_comment",
  "error_404",

  "script_present",
  "script_present_async",
  "script_present_async_false",
  "script_present_missing",
  "script_integrity_by_default",

  "stylesheet_present",
  "stylesheet_missing",

  "chunks_js_include",
  "chunks_css_include",
  "chunks_css_include_missing",
  "chunks_js_mixed",

  "chunks_css_include",
  "chunks_css_include_missing"
];

describe("asset_map", () => {
  const $ = cheerio.load(
    fs.readFileSync(path.resolve(__dirname, "../_site/index.html"), "utf8")
  );
  testIds.forEach(id => {
    it(id, () => {
      expect($(`#test--${id} .test-output`).text()).toMatchSnapshot();
    });
  });
});
