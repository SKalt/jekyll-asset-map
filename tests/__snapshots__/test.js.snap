// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asset_map chunks_css_include 1`] = `
"<link rel=\\"stylesheet\\" href=\\"app.chunk.1.css\\">
<link rel=\\"stylesheet\\" href=\\"app.chunk.2.css\\">
<link rel=\\"stylesheet\\" href=\\"app.chunk.3.css\\">
"
`;

exports[`asset_map chunks_css_include 2`] = `
"<link rel=\\"stylesheet\\" href=\\"app.chunk.1.css\\">
<link rel=\\"stylesheet\\" href=\\"app.chunk.2.css\\">
<link rel=\\"stylesheet\\" href=\\"app.chunk.3.css\\">
"
`;

exports[`asset_map chunks_css_include_missing 1`] = `
"<script>alert(\\"Warning! No entrypoint named missing found in manifest#entrypoints.\\")</script>
<link rel=\\"stylesheet\\" href=\\"app.chunk.1.css\\">
<link rel=\\"stylesheet\\" href=\\"app.chunk.3.css\\">
"
`;

exports[`asset_map chunks_css_include_missing 2`] = `
"<script>alert(\\"Warning! No entrypoint named missing found in manifest#entrypoints.\\")</script>
<link rel=\\"stylesheet\\" href=\\"app.chunk.1.css\\">
<link rel=\\"stylesheet\\" href=\\"app.chunk.3.css\\">
"
`;

exports[`asset_map chunks_js_include 1`] = `
"<script src=\\"app.chunk.1.js\\"></script>
<script src=\\"app.chunk.2.js\\"></script>
<script src=\\"app.chunk.3.js\\"></script>
"
`;

exports[`asset_map error_alert 1`] = `
"<script>alert(\\"this failed. Here's why:...\\")</script>
"
`;

exports[`asset_map error_comment 1`] = `
"<!-- this failed. Here's why:... -->
"
`;

exports[`asset_map error_error 1`] = `
"<script>throw new Error(\\"this failed. Here's why:...\\")</script>
"
`;

exports[`asset_map error_fail 1`] = `
"<script src=\\"/pack/does/not/exist/a-missing-pack.js\\"></script>
"
`;

exports[`asset_map integrity 1`] = `
"sha384-oqVuAfXRKap7fdgcCY5uykM6+R9GqQ8K/uxy9rx7HNQlGYl1kPzQho1wx4JwY8wC
"
`;

exports[`asset_map integrity_missing 1`] = `
"
"
`;

exports[`asset_map path_missing 1`] = `
"
"
`;

exports[`asset_map path_present 1`] = `
"foo.h4sh3d.js
"
`;

exports[`asset_map script_integrity_by_default 1`] = `
"<script src=\\"bar.md5hd.js\\" integrity=\\"sha384-oqVuAfXRKap7fdgcCY5uykM6+R9GqQ8K/uxy9rx7HNQlGYl1kPzQho1wx4JwY8wC\\"></script>
"
`;

exports[`asset_map script_present 1`] = `
"<script src=\\"foo.h4sh3d.js\\"></script>
"
`;

exports[`asset_map script_present_async 1`] = `
"<script src=\\"foo.h4sh3d.js\\" async></script>
"
`;

exports[`asset_map script_present_async_false 1`] = `
"<script src=\\"foo.h4sh3d.js\\"></script>
"
`;

exports[`asset_map script_present_missing 1`] = `
"<!-- WARNING! Missing script pack named quux.js -->
"
`;

exports[`asset_map stylesheet_missing 1`] = `
"<script>alert(\\"WARNING! missing stylesheet pack named missing.css\\")</script>
"
`;

exports[`asset_map stylesheet_present 1`] = `
"<link rel=\\"stylesheet\\" href=\\"foo.h45h3d.css\\">
"
`;
